src/main/java/org/duraspace/bagit/BagProfile.java

```diff
@@ -446,23 +443,61 @@ public Serialization getSerialization() {
      * @param config the BagConfig
      */
     public void validateConfig(final BagConfig config) {
-        for (final String section : sections) {
-            final String tagFile = section.toLowerCase() + ".txt";
-            if (config.hasTagFile(tagFile)) {
-                try {
-                    ProfileValidationUtil.validate(section, getMetadataFields(section),
-                                                   config.getFieldsForTagFile(tagFile));
+        checkRequiredTagsExist(config.getTagFiles());
+        for (final String section : config.getTagFiles()) {
+            validateTag(section, config.getFieldsForTagFile(section));
+        }
+    }
 
-                    ProfileValidationUtil.validateTagIsAllowed(Paths.get(tagFile), tagFilesAllowed);
-                } catch (ProfileValidationException e) {
-                    throw new RuntimeException(e.getMessage(), e);
-                }
-            } else {
-                throw new RuntimeException(String.format("Error missing section %s from bag config", section));
+    /**
+     * Test that all required tag files exist
+     *
+     * @param tags the name of each tag file to check
+     */
+    public void checkRequiredTagsExist(final Set<String> tags) {
```

Comment:
Both the `checkRequiredTagsExist` and `validateTag` methods were intended to be used with the Replication Task Suite. Since the RTS uses properties to load custom information, we're able to load custom values for tag files through its configuration, which maps to a similar structure found in the `BagConfig` of `Map<String, Map<String, String>>`. 

Though now I'm wondering if we should instead add a constructor for `BagConfig` which takes said `Map` instead of adding these additional methods in `BagProfile`.

Reply:
Based on discussion with @bbranan, we decided to have a single public method `validateTagFiles` which takes a `Map<String, Map<String, String>>` to be used for validation. This better fits the workflow we already have with the `BagConfig` and loc `Bag`s.

The two outlined methods are now private.